docs/src/rules/no-restricted-imports.md

```diff
@@ -237,137 +202,147 @@ import * as Foo from "foo";
 
 :::
 
-::: incorrect { "sourceType": "module" }
+Examples of **correct** code for `importNames` in `paths`:
+
+Other strings doesn't report default exported import even if the names are same:
+
+::: correct { "sourceType": "module" }
 
 ```js
-/*eslint no-restricted-imports: ["error", { patterns: [{
-    group: ["lodash/*"],
-    message: "Please use the default import from 'lodash' instead."
-}]}]*/
+/*eslint no-restricted-imports: ["error", { paths: [{ name: "foo", importNames: ["DisallowedObject"] }] }]*/
 
-import pick from 'lodash/pick';
+import DisallowedObject from "foo"
 ```
 
 :::
 
-::: incorrect { "sourceType": "module" }
+::: correct { "sourceType": "module" }
 
 ```js
-/*eslint no-restricted-imports: ["error", { patterns: [{
-    group: ["foo[A-Z]*"],
-    caseSensitive: true
+/*eslint no-restricted-imports: ["error", { paths: [{
+    name: "foo",
+    importNames: ["DisallowedObject"],
+    message: "Please import 'DisallowedObject' from '/bar/baz/' instead."
 }]}]*/
 
-import pick from 'fooBar';
+import { AllowedObject as DisallowedObject } from "foo";
 ```
 
 :::
 
-::: incorrect { "sourceType": "module" }
+### patterns
 
-```js
-/*eslint no-restricted-imports: ["error", { patterns: [{
-    group: ["utils/*"],
-    importNames: ['isEmpty'],
-    message: "Use 'isEmpty' from lodash instead."
-}]}]*/
+This is also an object option whose value is an array. This option comes handy when you need to restrict the modules present inside a perticular directory. Simply, you can specify the modules inside the directories you want to restrict as string in `gitignore-style` patterns.
 
-import { isEmpty } from 'utils/collection-utils';
+Since the patterns follows the `gitignore-style` then if you want to reinclude any particular module this can be done by prefixing a negation (`!`) mark in front of the pattern and it's good to keep the negation patterns in last as `gitignore-style` follows order:
```

Comment:
```suggestion
Because the patterns follow `gitignore`-style, if you want to reinclude any particular module this can be done by prefixing a negation (`!`) mark in front of the pattern. (Negated patterns should come last in the array because order is important.)
```